go/arrow/avro/schema.go

```diff
@@ -0,0 +1,407 @@
+				avroLogicalToArrowField(c)
+			} else {
+				c := n.newChild(f.Name(), f.Type())
+				arrowSchemafromAvro(c)
+			}
+
+		}
+	}
+	var fields []arrow.Field
+	for _, child := range n.Children() {
+		fields = append(fields, child.arrowField)
+	}
+
+	namedSchema, ok := isNamedSchema(n.schema)
+	if !ok || namedSchema == n.name+"_data" || !n.union {
+		n.arrowField = arrow.Field{Name: n.name, Type: arrow.StructOf(fields...), Nullable: true}
+	} else {
+		n.arrowField = arrow.Field{Name: n.name, Type: arrow.StructOf(fields...), Nullable: true, Metadata: arrow.MetadataFrom(map[string]string{"typeName": namedSchema})}
+	}
+
+}
+
+func isLogicalSchemaType(s avro.Schema) bool {
+	lts, ok := s.(avro.LogicalTypeSchema)
+	if !ok {
+		return false
+	}
+	if lt := lts.Logical(); lt != nil {
+		return true
+	}
+	return false
+}
+
+func isNamedSchema(s avro.Schema) (string, bool) {
+	if ns, ok := s.(avro.NamedSchema); ok {
+		return ns.FullName(), ok
+	}
+	return "", false
+}
+
+// Avro primitive type.
+//
+// NOTE: Arrow Binary type is used as a catchall to avoid potential data loss.
+func avroPrimitiveToArrowType(avroFieldType string) arrow.DataType {
+	switch avroFieldType {
+	// int: 32-bit signed integer
+	case "int":
+		return arrow.PrimitiveTypes.Int32
+	// long: 64-bit signed integer
+	case "long":
+		return arrow.PrimitiveTypes.Int64
+	// float: single precision (32-bit) IEEE 754 floating-point number
+	case "float":
+		return arrow.PrimitiveTypes.Float32
+	// double: double precision (64-bit) IEEE 754 floating-point number
+	case "double":
+		return arrow.PrimitiveTypes.Float64
+	// bytes: sequence of 8-bit unsigned bytes
+	case "bytes":
+		return arrow.BinaryTypes.Binary
+	// boolean: a binary value
+	case "boolean":
+		return arrow.FixedWidthTypes.Boolean
+	// string: unicode character sequence
+	case "string":
+		return arrow.BinaryTypes.String
+	// fallback to binary type for any unsupported type
+	// (shouldn't happen as all types specified in avro 1.11.1 are supported)
+	default:
+		return arrow.BinaryTypes.Binary
+	}
+}
+
+func avroLogicalToArrowField(n *schemaNode) arrow.Field {
+	var dt arrow.DataType
+	// Avro logical types
+	switch lt := n.schema.(avro.LogicalTypeSchema).Logical(); lt.Type() {
+	// The decimal logical type represents an arbitrary-precision signed decimal number of the form unscaled × 10-scale.
+	// A decimal logical type annotates Avro bytes or fixed types. The byte array must contain the two’s-complement
+	// representation of the unscaled integer value in big-endian byte order. The scale is fixed, and is specified
+	// using an attribute.
+	//
+	// The following attributes are supported:
+	// scale, a JSON integer representing the scale (optional). If not specified the scale is 0.
+	// precision, a JSON integer representing the (maximum) precision of decimals stored in this type (required).
+	case "decimal":
+		id := arrow.DECIMAL128
+		if lt.(*avro.DecimalLogicalSchema).Precision() > 38 {
```

Comment:
use the constant, `decimal128.MaxPrecision`

Reply:
changed to use constant in comparison